src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthDirectives.scala

```diff
@@ -23,24 +24,50 @@ trait OAuthDirectives {
   def authenticationToken(requestDescription: String): Directive1[String] =
     headerValue(authorization()) | reject(CredentialsMissingRejection(requestDescription))
 
-  def authenticate(token: String, requestDescription: String, authService: AuthService): Directive1[AuthenticatedUser] =
-    authService.authenticate(token) match {
-      case Success(authUser) => provide(authUser)
-      case Failure(ex) => {
-        logger.error(s"Authentication failed with exception: ${ex.getMessage}")
-        reject(CredentialsRejectedRejection(requestDescription))
+  def authenticate(token: String, requestDescription: String, authUserToRoute: AuthenticatedUser => Route)(implicit authService: AuthService, executionContext: ExecutionContext): Route = {
```

Comment:
it is implicit...for methods you only put one implicit keyword... per parameter list... :walking: